packages/context/src/binding.ts

```diff
@@ -328,11 +349,26 @@ export class Binding<T = BoundValue> {
    */
   applyDefaultScope(scope: BindingScope): this {
     if (!this._scope) {
-      this._scope = scope;
+      this.inScope(scope);
     }
     return this;
   }
 
+  /**
+   * Set the `_getValue` function
+   * @param getValueFn getValue function
+   */
+  private _setValueGetter(
+    getValueFn: (
+      ctx: Context,
+      optionsOrSession?: ResolutionOptionsOrSession,
```

Comment:
What is the reasoning for asking every value getter variant to support both flavors of `options` and `session`? From what I see in other parts of this patch, it adds unnecessary complexity to getter implementations.

IMO, `ResolutionOptionsOrSession` should be accepted only by `getValue` method for backwards compatibility. Under the hood, `getValue` should "normalize" that argument to `ResolutionOptions` and use that options in all places going forward.